src/helpers/blood-pressure-data-providers/device-blood-pressure-data-provider.ts

```diff
@@ -0,0 +1,39 @@
+import { DeviceDataNamespace, DeviceDataPoint } from "@careevolution/mydatahelps-js";
+import queryAllDeviceData from "../daily-data-providers/query-all-device-data";
+import { BloodPressureDataPoint } from ".";
+import { parseISO, startOfDay } from "date-fns";
+
+export default async function (namespace: DeviceDataNamespace, bloodPressureSystolic: string, bloodPressureDiastolic: string): Promise<BloodPressureDataPoint[]> {
+
+    function collateResults(dataPoints: DeviceDataPoint[]): BloodPressureDataPoint[] {
+        let bpDataPointsByObservationDateTime: Map<string, BloodPressureDataPoint> = new Map<string, BloodPressureDataPoint>();
+
+        dataPoints.forEach((dataPoint) => {
+            if (dataPoint.observationDate && !isNaN(Number(dataPoint.value))) {
+                var bpDate = parseISO(dataPoint.observationDate);
+                var bpDateIdentifier = bpDate.toDateString();
+                var exists = bpDataPointsByObservationDateTime.get(bpDateIdentifier);
+                var bp: BloodPressureDataPoint = exists ?? { date: startOfDay(bpDate), systolic: 0, diastolic: 0 };
```

Comment:
What if there is not a matching systolic for a diastolic reading?  It is not likely but possible so we should probably just discard that

Reply:
Addressed